internal/commands/server/create.go

```diff
@@ -303,11 +306,9 @@ func (s *createCommand) ExecuteWithoutArguments(exec commands.Executor) (output.
 			return nil, fmt.Errorf("both --cores and --memory must be defined for custom plans")
 		}
 
-		if s.params.Plan != "" && s.params.Plan != customPlan {
-			return nil, fmt.Errorf("--plan needs to be 'custom' or omitted when --cores and --memory are specified")
+		if s.params.Plan != customPlan {
+			return nil, fmt.Errorf("--plan needs to be 'custom' when --cores and --memory are specified")
 		}
-
-		s.params.Plan = customPlan // Valid for all custom plans.
```

Comment:
This is not really needed, because `Plan` has default value. So `s.params.Plan` will always be either `custom` or a simple plan string